frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -48,22 +54,65 @@ export const surveysLogic = kea<surveysLogicType>([
         setSearchTerm: (searchTerm: string) => ({ searchTerm }),
         setSurveysFilters: (filters: Partial<SurveysFilters>, replace?: boolean) => ({ filters, replace }),
         setTab: (tab: SurveysTabs) => ({ tab }),
+        loadNextPage: true,
+        loadBackendSearchResults: true,
     }),
     loaders(({ values }) => ({
         surveys: {
-            __default: [] as Survey[],
+            __default: { results: [], count: 0 } as CountedPaginatedResponse<Survey>,
             loadSurveys: async () => {
-                const responseSurveys = await api.surveys.list()
-                return responseSurveys.results
+                return await api.surveys.list()
+            },
+            loadNextPage: async () => {
+                if (!values.surveys.next || !isURL(values.surveys.next)) {
+                    return values.surveys
+                }
+
+                const url = new URL(values.surveys.next)
+                const limit = parseInt(url.searchParams.get('limit') || SURVEY_PAGE_SIZE.toString())
+                const offset = parseInt(url.searchParams.get('offset') || values.surveys.results.length.toString())
+
+                const response = await api.surveys.list({ limit, offset })
+
+                // deduplicate results
+                const existingIds = new Set(values.surveys.results.map((s) => s.id))
+                const newResults = response.results.filter((s) => !existingIds.has(s.id))
```

Comment:
why do we need to dedupe here? if the API is called with the right offset and limit, you should just append new results and stop loading once its finished

Reply:
because we might have search for other elements that are not included on the page, and if we don't dedupe, it's possible they will show up in the other pages then appear on the list twice

Reply:
this is due to the implementation then, the search results could be a different 'list', but yeah no strong preferences here, your call

Reply:
now we have separate logic for the search results